cogs/embeds.py

```diff
@@ -1,24 +1,142 @@
 import discord
 from discord.ext import commands
+import json
+import re
 
 
 class Embeds(commands.Cog):
+
     def __init__(self, bot: commands.Bot) -> None:
         self.bot = bot
-    
+
     @commands.hybrid_command()
     async def embed(self, ctx: commands.Context):
         """
         Interactively construct a Discord embed.
         """
-        raise NotImplementedError('Command requires implementation and permission set-up.')
-    
+
+        def check(m: discord.Message):
+            return m.channel == ctx.channel and m.author == ctx.author
+
+        def is_skip(msg: discord.Message):
+            return msg.content.strip().lower() == "skip"
+
+        def is_done(msg: discord.Message):
+            return msg.content.strip().lower() == "done"
+
+
+        async def handle_color(embed: discord.Embed):
+            while True:
+                try:
+                    color_msg: discord.Message = await self.bot.wait_for("message", check=check, timeout=timeout)
+                    if is_skip(color_msg):
+                        break
+                    color_str = color_msg.content
+                    embed.color = discord.Color.from_str(color_str)
+                    break
+                except:
```

Comment:
It is good practice to always `except` specific exceptions or errors. The `try` clause can raise [asyncio.TimeoutError](https://discordpy.readthedocs.io/en/stable/ext/commands/api.html?highlight=wait_for#discord.ext.commands.Bot.wait_for) or [ValueError](https://discordpy.readthedocs.io/en/stable/api.html?highlight=from_str#discord.Colour.from_str), so it might be good to deal with these cases individually.
```py
try:
    ...
except ValueError as e:
    ...  # handle invalid color
except asyncio.TimeoutError as e:
    ...  # handle timeout
```
By not having a general `except` statement, errors that aren't foreseen will be thrown (which is preferred behavior, else they will be `except`ed silently).

Reply:
How do we handle those currently? I'll change it to that.

Reply:
For an invalid color string, the following is sent:
> Invalid format. Please use hex, for example: #123456.

When the command times out, the following is sent:
> Input timed out.

As long as these cases are dealt with separately it should be fine.